pkg/command/root.go

```diff
@@ -125,9 +127,12 @@ func newRootCmd() *cobra.Command {
 			//   - 'config set' (would be a chicken and egg issue if user tries to set CEIP configuration
 			//      using "tanzu config set env.TANZU_CLI_CEIP_OPT_IN_PROMPT_ANSWER yes")
 			//   - 'config eula xxx', 'ceip set' (auto prompting when running these commands is confusing)
+			//   - 'tanzu pinniped-auth' ( since this command is being invoked by kubectl exec plugin, user doesn't
+			//      get to see the prompts and kubectl command execution just gets stuck and it is very hard for users to figure out)
 			if cmd.Name() != cobra.ShellCompRequestCmd && cmd.Name() != completionCmd.Name() &&
 				cmd.Name() != newVersionCmd().Name() &&
-				!(cmd.Name() == cmdNameSet && cmd.Parent().Name() == configCmd.Name()) {
+				!(cmd.Name() == cmdNameSet && cmd.Parent().Name() == configCmd.Name()) &&
+				!(cmd.Name() == cmdNamePinnipedAuth && cmd.Parent().Name() == cmdNameTanzu) {
```

Comment:
See next comment first as it may cancel this one.

--

Can a command called "pinniped-auth" be called in any other situation than `tanzu pinniped-auth`?
It may be superfluous to check that the parent is `tanzu`, in which case we don't need to create the `cmdNameTanzu` constant.
I'm asking because if we are to create a constant `cmdNameTanzu` for the name of the root command, we should probably use it at line 114 above also.

Reply:
Updated the PR as per the next comment. So it is not applicable anymore.